tests/nutanix/test_functional_cloudinit.py

```diff
@@ -912,6 +912,124 @@ def test_cloudinit_no_networkmanager(self):
             "cloud-final.service status is not active (exited)")
         self.session.cmd_output("sudo subscription-manager unregister")
 
+    def _generate_password(self, password, hash, salt=''):
+        import crypt
+        if hash == 'md5':
+            crypt_type = '$1$'
+        elif hash == 'sha-256':
+            crypt_type = '$5$'
+        elif hash == 'sha-512':
+            crypt_type = '$6$'
+        else:
+            assert False, 'Unhandled hash option: {}'.format(hash)
+        # Generate a random salt
+        if salt == '':
+            with open('/dev/urandom', 'rb') as urandom:
+                while True:
+                    byte = urandom.read(1)
+                    byte = byte.decode("gbk","ignore")
```

Comment:
What is this for? why we need to handle gbk encode here?

Reply:
I need to change the type for variable byte from byte to string. I use UTF-8 at first, but I met an error caused by a special random byte, then I change to gbk to resovle the error.

Reply:
Let's not decode it here, you can compare with byte string next line like (b'ABC'), and then decode with utf8 afterwards when concatenate with var salt.

Reply:
fixed